surfinBH/_fit_evaluators/fit_7dq4Emri.py

```diff
@@ -0,0 +1,267 @@
+        chiBmag= np.random.uniform(0, self.hard_param_lims['chiBmag'])
+        if self.aligned_spin_only:
+            chiAph, chiBph = 0, 0
+            chiAth, chiBth = np.random.choice([0, np.pi]), \
+                np.random.choice([0, np.pi])
+        else:
+            chiAth = np.arccos(np.random.uniform(-1., 1.))
+            chiBth = np.arccos(np.random.uniform(-1., 1.))
+            chiAph = np.random.uniform(0, 2*np.pi)
+            chiBph = np.random.uniform(0, 2*np.pi)
+
+        chiA = [chiAmag*np.sin(chiAth)*np.cos(chiAph),
+                chiAmag*np.sin(chiAth)*np.sin(chiAph),
+                chiAmag*np.cos(chiAth)]
+
+        chiB = [chiBmag*np.sin(chiBth)*np.cos(chiBph),
+                chiBmag*np.sin(chiBth)*np.sin(chiBph),
+                chiBmag*np.cos(chiBth)]
+
+        return q, chiA, chiB
+
+    #-------------------------------------------------------------------------
+    def _eval_wrapper(self, fit_key, q, chiA, chiB, **kwargs):
+        """ Evaluates the NRSur7dq4EmriRemnant model. """
+        chiA = np.array(chiA)
+        chiB = np.array(chiB)
+
+        # Warn/Exit if extrapolating
+        allow_extrap = kwargs.pop('allow_extrap', False)
+        if allow_extrap: warnings.warn("""Optional argument 'allow_extrap' is 
+        unused for this model. It works at arbitrarily large mass ratios.""")
+        self._check_param_limits(q, chiA, chiB, True)
+
+        x = np.concatenate(([q], chiA, chiB))
+
+        def eval_r_isco(chiz):
+            """
+            Estimate ISCO radius following Eq. (2.21) by Bardeen et al. 
+            (https://doi.org/10.1086/151796)
+            """
+ 
+            Z1 = 1+(1-chiz**2)**(1/3)*((1+chiz)**(1/3)+(1-chiz)**(1/3))
+            Z2 = (3*chiz**2+Z1**2)**(1/2)
+            r_ISCO = 3+Z2-np.sign(chiz)*((3-Z1)*(3+Z1+2*Z2))**(1/2)
+    
+            return r_ISCO
+
+        def eval_emriChif(q, chiA):
+            """
+            Estimate remnant spin for EMRI binary following Eq. (7) 
+            presented in THE PAPER.    
+            """
+            r_ISCO = eval_r_isco(chiA[2])
+            E_ISCO = np.sqrt(1-2/(3*r_ISCO))
+            L_ISCO = np.array([0,0,2/(3*3**(1/2))*(1+2*np.sqrt(3*r_ISCO-2))])
+                      
+            return chiA + 1/q*(L_ISCO-2*chiA*E_ISCO)  
+
+        def extrap_chif(x, fit_key):
+            """
+            The model evaluates remnant spin using GPR fit up to the upper bound
+            of the EMRI training region (q = 1000). At q > 1000, it returns the 
+            expected EMRI limit (Eq. (7) in THE PAPER). There is a transition 
+            function (Eq. (9) in THE PAPER) connecting smoothly these 
+            two regions. The transition takes place from q=1000 to q=2000.
+            """
+            x_i = x.copy()
+            x_i[0] = self.qEmriMax
+            
+            """This if statement defines the width of the transition region"""
+            if x[0] < 2*self.qEmriMax:
```

Comment:
In the paper you say 
<img width="212" alt="image" src="https://github.com/vijayvarma392/surfinBH/assets/9355891/48966891-06e5-4443-8a4e-b38b11ab06b1">
but here it looks like you are doing this for q below 2*qMax?

Reply:
I have to make a change in the paper. There is no point in applying the transition at the extrema, it simply computes the value with the GPR fit or the EMRI limit (but with useless additional steps)